File: back-end/Proyecto.Domain/Models/Response/PagedResponse.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Proyecto.Domain.Models.Response
{
    public class PagedResponse<T>
    {
		public IEnumerable<T> Data { get; set; }
		public int? PageNumber { get; set; }
		public int? PageSize { get; set; }
		public int? MaxPage { get; set; }
		public int? Total { get; set; }
		public string NextPage { get; set; }
		public string PreviousPage { get; set; }
		
		public PagedResponse()
        {

        }

		public PagedResponse(IEnumerable<T> data)
		{
			Data = data;
		}
	}
}

File: front-end/src/app/models/PagedResponse.ts
import { GrupoGraduadoResponse } from './GrupoGraduado';

export interface GrupoGraduadoPagedResponse {
	data: GrupoGraduadoResponse[];
	pageNumber: number;
	pageSize: number;
	maxPage: number;
	total: number;
	nextPage: string;
	previousPage: string;
}
